src/shared/tpm2-util.h

```diff
@@ -66,7 +66,25 @@ static inline void tpm2_context_freep(struct tpm2_context **c) {
                 *c = tpm2_context_free(*c);
 }
 
-ESYS_TR tpm2_flush_context_verbose(struct tpm2_context *c, ESYS_TR handle);
+struct tpm2_handle {
+        struct tpm2_context *context;
+        ESYS_TR handle;
+};
+
+#define _tpm2_handle_init(c, h) (struct tpm2_handle) { .context = (c), .handle = (h), }
+#define tpm2_handle_init(c) _tpm2_handle_init(c, ESYS_TR_NONE)
+struct tpm2_handle tpm2_handle_release(struct tpm2_handle h);
+static inline void tpm2_handle_releasep(struct tpm2_handle *h) {
+        if (h)
+                *h = tpm2_handle_release(*h);
+}
+
+#define HANDLE_NONE tpm2_handle_init(NULL)
+#define HANDLE_RH_OWNER _tpm2_handle_init(NULL, ESYS_TR_RH_OWNER)
+#define HANDLE_PASSWORD _tpm2_handle_init(NULL, ESYS_TR_PASSWORD)
```

Comment:
please prefix with TPM2 as usual.

make call these TPM2_HANLE_MAKE_RH_OWNER() and so on?

Reply:
ack, the RH_OWNER and _PASSWORD aren't needed yet so i'll drop those, and update the _NONE macro